sympy/solvers/solveset.py

```diff
@@ -323,7 +323,7 @@ def inv(trig):
 def _invert_complex(f, g_ys, symbol):
     """Helper function for _invert."""
 
-    if f == symbol:
+    if f == symbol or g_ys is S.EmptySet:
         return (f, g_ys)
```

Comment:
If you do this, symbol won't be isolated from `f`. Perhaps the return should be `(symbol, g_ys)`

Reply:
It doesn't matter. Exactly the same thing will happen if none of the if-cases below is true.